src/Tempest/Http/src/RouteConfig.php

```diff
@@ -29,7 +34,14 @@ public function addRoute(MethodReflector $handler, Route $route): self
         if ($route->isDynamic) {
             $this->regexMark = str_increment($this->regexMark);
             $this->dynamicRoutes[$route->method->value][$this->regexMark] = $route;
-            $this->addToMatchingRegex($route, $this->regexMark);
+
+            $this->routingTree->add(
```

Comment:
So this is the actual start of the whole new routing structure. Let me explain what the whole idea is:

Instead of "just appending" route regexes one after the other. I want to build up a hierarchical regex, where groups of route with common prefixes are matches together with the same prefix. So if we got 2 routes `/test/edit` and `/test/delete` , previously we would create a regex `~^/test/edit(*MARK:b)$|^/test/delete(*MARK:c)$~` While we want to create `~^/test/(?edit(*MARK:b)$|delete(*MARK:c)$)~` (these regexes are written by hand, so if I made an small error, sorry, but you should get the idea.)

A good way to  get this route hierarchy is to create a routing tree first, and than create the regex as a product. After we created the matching regex we also do not need the tree anymore. And building up trees can be efficient enough.